x-pack/plugins/security_solution/server/lib/detection_engine/rules/bulk_action_update.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { RulesSchema } from '../../../../common/detection_engine/schemas/response/rules_schema';
+import {
+  BulkActionUpdatePayload,
+  BulkActionUpdateType,
+} from '../../../../common/detection_engine/schemas/common/schemas';
+
+export const addItemsToArray = <T>(arr: T[], items: T[]): T[] =>
+  Array.from(new Set([...arr, ...items]));
+
+export const deleteItemsFromArray = <T>(arr: T[], items: T[]): T[] => {
+  const itemsSet = new Set(items);
+  return arr.filter((item) => !itemsSet.has(item));
+};
+
+export const appplyBulkActionUpdateToRule = (
+  existingRule: Partial<RulesSchema>,
+  action: BulkActionUpdatePayload
+): Partial<RulesSchema> => {
```

Comment:
`RulesSchema` is the schema used in the API responses, and I think we should use it only to populate the responses. I would keep business logic like this function dependent on internal types like Alerting Framework's types, SO types, SO parameters etc. This would allow us to modularize the codebase and keep API schemas separate from the core domain logic (which is currently not super explicit and lives in routes, executors and all sorts of utilities).

Specifically, I think this function could depend on `RuleAlertType` exported from `x-pack/plugins/security_solution/server/lib/detection_engine/rules/types.ts`.

Reply:
renamed as `edit`

Reply:
Implemented: I'm passing now `RuleAlertType`.